src/core/lib/security/credentials/external/url_external_account_credentials.cc

```diff
@@ -120,8 +114,19 @@ void UrlExternalAccountCredentials::RetrieveSubjectToken(
   cb_ = cb;
   grpc_httpcli_request request;
   memset(&request, 0, sizeof(grpc_httpcli_request));
-  request.host = const_cast<char*>(url_.authority().c_str());
-  request.http.path = gpr_strdup(url_.path().c_str());
+  absl::StatusOr<URI> tmp_url = URI::Parse(url_);
+  if (!tmp_url.ok()) {
+    FinishRetrieveSubjectToken(
+        "", GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+                absl::StrFormat("Invalid credential source url.").c_str()));
+    return;
+  }
+  URI url_object = *tmp_url;
+  request.host = const_cast<char*>(url_object.authority().c_str());
+  std::vector<std::string> v =
+      absl::StrSplit(url_.c_str(), absl::MaxSplits('/', 3));
```

Comment:
A few things here, in order of severity:

What is this assuming about the structure of the URI string? I imagine the assumption is something like `<scheme>://<authority>/<path>...`, but can we assume that all external account credential URIs are in that format? For example, URIs like `http://foo.com?query` (no path), `dns:foo.com` and `ipv6:[d00d:90%eth1]` are all valid, but I'm not sure they'll ever be used here.

absl::StrSplit can take an std::string, no need to retrieve the c_str.

You cn (I believe) create an std::vector<absl::string_view> here instead and save on string copies.

Reply:
For url creds, it's assumed that the url has a static format of `<scheme>://<authority>/<path>`. It's very unlikely to change but it seems too big an assumption given the cases you provide.

I'm trying to use the class `URI` to parse the url, while there might be some issue. Please see my comments in the other thread.

Reply:
That's fair. I'll defer to you on making the judgement call of which set of parsing issues to deal with :-). Also note, this string split method will capture the URI fragment as well.